### ISSUE TITLE
blobfuse uploaded the same file twice to Storage Account when using SCP.

### ISSUE THREAD

**Issue by takmnagayams** [Azure/azure-storage-fuse#523]
Can I suggest a feature that allows files to be uploaded once to Storage Account when using scp?

### Which version of the blobfuse was used?
1.0.2

### Which OS (please include version) are you using?
istributor ID: Ubuntu
Description:    Ubuntu 18.04.5 LTS
Release:        18.04
Codename:       bionic

### What problem was encountered?
When I uploaded a file with scp command, the file was uploaded to the Storage Account twice as following logs.
I didn’t update the file which was uploaded, but it was uploaded to the Storage Account twice.

### Have you found a mitigation/solution?
No

### By default, blobfuse logs errors to syslog. If this is relevant, is there anything in the syslog that might be helpful?

```
Oct 26 13:26:49 blobfuse systemd[1]: Started Session 522 of user tanagaya.
Oct 26 13:26:50 blobfuse blobfuse[2796]: Function azs_getattr, in file /home/vsts/work/1/s/blobfuse/utilities.cpp, line 120: azs_getattr called with path = /tt7.png
Oct 26 13:26:50 blobfuse blobfuse[2796]: Function azs_getattr, in file /home/vsts/work/1/s/blobfuse/utilities.cpp, line 169: Object /mnt/resource/blobfusetmp/root/tt7.png is not in the local cache during getattr.
Oct 26 13:26:50 blobfuse blobfuse[2796]:  ==> REQUEST/RESPONSE :: GET https://tanagayablobfuse.blob.core.windows.net/mycontainer?comp=list&delimiter=/&include=metadata&maxresults=5000&prefix=tt7.png&restype=container?&User-Agent=azure-storage-fuse/1.3.4&x-ms-date=Mon, 26 Oct 2020 13:26:50 GMT&x-ms-version=2018-11-09&Authorization=****&Transfer-Encoding=--------------------------------------------------------------------------------RESPONSE Status :: 200 :: REQ ID : 3afdcc3b-d01e-009a-179b-ab0134000000
Oct 26 13:26:50 blobfuse blobfuse[2796]: Function azs_getattr, in file /home/vsts/work/1/s/blobfuse/utilities.cpp, line 233: In azs_getattr list_segmented_item do loop blob prefix: tt7.png continuation:
Oct 26 13:26:50 blobfuse blobfuse[2796]: Function azs_getattr, in file /home/vsts/work/1/s/blobfuse/utilities.cpp, line 342: tt7.png not returned in list_segmented_blobs. It is a new blob
Oct 26 13:26:50 blobfuse blobfuse[2796]: Function azs_create, in file /home/vsts/work/1/s/blobfuse/fileapis.cpp, line 193: azs_create called with path = /tt7.png, mode = 33277, fi->flags = 8041
Oct 26 13:26:50 blobfuse blobfuse[2796]: Function ensure_files_directory_exists_in_cache, in file /home/vsts/work/1/s/blobfuse/utilities.cpp, line 96: Making cache directory /mnt.
Oct 26 13:26:50 blobfuse blobfuse[2796]: Function ensure_files_directory_exists_in_cache, in file /home/vsts/work/1/s/blobfuse/utilities.cpp, line 96: Making cache directory /mnt/resource.
Oct 26 13:26:50 blobfuse blobfuse[2796]: Function ensure_files_directory_exists_in_cache, in file /home/vsts/work/1/s/blobfuse/utilities.cpp, line 96: Making cache directory /mnt/resource/blobfusetmp.
Oct 26 13:26:50 blobfuse blobfuse[2796]: Function ensure_files_directory_exists_in_cache, in file /home/vsts/work/1/s/blobfuse/utilities.cpp, line 96: Making cache directory /mnt/resource/blobfusetmp/root.
Oct 26 13:26:50 blobfuse blobfuse[2796]: Successfully created file /tt7.png in file cache.
Oct 26 13:26:50 blobfuse blobfuse[2796]: Function azs_create, in file /home/vsts/work/1/s/blobfuse/fileapis.cpp, line 225: Returning success from azs_create with file /tt7.png.
Oct 26 13:26:50 blobfuse blobfuse[2796]: Function azs_getattr, in file /home/vsts/work/1/s/blobfuse/utilities.cpp, line 120: azs_getattr called with path = /tt7.png
Oct 26 13:26:50 blobfuse blobfuse[2796]: Function azs_getattr, in file /home/vsts/work/1/s/blobfuse/utilities.cpp, line 152: Accessing mntPath = /mnt/resource/blobfusetmp/root/tt7.png for getattr succeeded; object is in the local cache.
Oct 26 13:26:50 blobfuse blobfuse[2796]: Function azs_getattr, in file /home/vsts/work/1/s/blobfuse/utilities.cpp, line 163: lstat on file /mnt/resource/blobfusetmp/root/tt7.png in local cache succeeded.
Oct 26 13:26:51 blobfuse blobfuse[2796]: Function azs_truncate, in file /home/vsts/work/1/s/blobfuse/fileapis.cpp, line 473: azs_truncate called.  Path = /tt7.png, offset = 19419
Oct 26 13:26:51 blobfuse blobfuse[2796]: azs_open called with path = /tt7.png, fi->flags = 2.
Oct 26 13:26:51 blobfuse blobfuse[2796]: Function azs_open, in file /home/vsts/work/1/s/blobfuse/fileapis.cpp, line 132: Opening /mnt/resource/blobfusetmp/root/tt7.png gives fh = 26, errno = 0
Oct 26 13:26:51 blobfuse blobfuse[2796]: Function azs_open, in file /home/vsts/work/1/s/blobfuse/fileapis.cpp, line 161: Returning success from azs_open, file = /tt7.png
Oct 26 13:26:51 blobfuse blobfuse[2796]: Function azs_truncate, in file /home/vsts/work/1/s/blobfuse/fileapis.cpp, line 498: Successfully truncated file /mnt/resource/blobfusetmp/root/tt7.png in the local file cache.
Oct 26 13:26:51 blobfuse blobfuse[2796]: Function azs_flush, in file /home/vsts/work/1/s/blobfuse/fileapis.cpp, line 259: azs_flush called with path = /tt7.png, fi->flags = 2, (((struct fhwrapper *)fi->fh)->fh) = 26.
Oct 26 13:26:51 blobfuse blobfuse[2796]: Function azs_flush, in file /home/vsts/work/1/s/blobfuse/fileapis.cpp, line 276: Successfully looked up mntPath.  Input path = /tt7.png, path_link_buffer = /proc/self/fd/26, path_buffer = /mnt/resource/blobfusetmp/root/tt7.png
Oct 26 13:26:51 blobfuse blobfuse[2796]:  ==> REQUEST/RESPONSE :: PUT https://tanagayablobfuse.blob.core.windows.net/mycontainer/tt7.png?&Content-Length=19419&x-ms-blob-type=BlockBlob&User-Agent=azure-storage-fuse/1.3.4&x-ms-date=Mon, 26 Oct 2020 13:26:51 GMT&x-ms-version=2018-11-09&Authorization=****&Transfer-Encoding=--------------------------------------------------------------------------------RESPONSE Status :: 201 :: REQ ID : 031581ae-e01e-005a-329b-abf90a000000
Oct 26 13:26:51 blobfuse blobfuse[2796]: Successfully uploaded file /tt7.png to blob tt7.png.
Oct 26 13:26:51 blobfuse blobfuse[2796]: Function azs_release, in file /home/vsts/work/1/s/blobfuse/fileapis.cpp, line 366: azs_release called with path = /tt7.png, fi->flags = 2
Oct 26 13:26:51 blobfuse blobfuse[2796]: Function azs_release, in file /home/vsts/work/1/s/blobfuse/fileapis.cpp, line 387: Adding file to the GC from azs_release.  File = /mnt/resource/blobfusetmp/root/tt7.png#012.
Oct 26 13:26:51 blobfuse blobfuse[2796]: Function azs_getattr, in file /home/vsts/work/1/s/blobfuse/utilities.cpp, line 120: azs_getattr called with path = /tt7.png
Oct 26 13:26:51 blobfuse blobfuse[2796]: Function azs_getattr, in file /home/vsts/work/1/s/blobfuse/utilities.cpp, line 152: Accessing mntPath = /mnt/resource/blobfusetmp/root/tt7.png for getattr succeeded; object is in the local cache.
Oct 26 13:26:51 blobfuse blobfuse[2796]: Function azs_getattr, in file /home/vsts/work/1/s/blobfuse/utilities.cpp, line 163: lstat on file /mnt/resource/blobfusetmp/root/tt7.png in local cache succeeded.
Oct 26 13:26:51 blobfuse blobfuse[2796]: Function azs_flush, in file /home/vsts/work/1/s/blobfuse/fileapis.cpp, line 259: azs_flush called with path = /tt7.png, fi->flags = 0, (((struct fhwrapper *)fi->fh)->fh) = 25.
Oct 26 13:26:51 blobfuse blobfuse[2796]: Function azs_flush, in file /home/vsts/work/1/s/blobfuse/fileapis.cpp, line 276: Successfully looked up mntPath.  Input path = /tt7.png, path_link_buffer = /proc/self/fd/25, path_buffer = /mnt/resource/blobfusetmp/root/tt7.png
Oct 26 13:26:51 blobfuse blobfuse[2796]:  ==> REQUEST/RESPONSE :: PUT https://tanagayablobfuse.blob.core.windows.net/mycontainer/tt7.png?&Content-Length=19419&x-ms-blob-type=BlockBlob&User-Agent=azure-storage-fuse/1.3.4&x-ms-date=Mon, 26 Oct 2020 13:26:51 GMT&x-ms-version=2018-11-09&Authorization=****&Transfer-Encoding=--------------------------------------------------------------------------------RESPONSE Status :: 201 :: REQ ID : 8d706abe-b01e-001a-429b-abfe32000000
Oct 26 13:26:51 blobfuse blobfuse[2796]: Successfully uploaded file /tt7.png to blob tt7.png.
Oct 26 13:26:51 blobfuse blobfuse[2796]: Function azs_release, in file /home/vsts/work/1/s/blobfuse/fileapis.cpp, line 366: azs_release called with path = /tt7.png, fi->flags = 32769
Oct 26 13:26:51 blobfuse blobfuse[2796]: Function azs_release, in file /home/vsts/work/1/s/blobfuse/fileapis.cpp, line 387: Adding file to the GC from azs_release.  File = /mnt/resource/blobfusetmp/root/tt7.png#012.
```

### If relevant, please share your mount command.
sudo blobfuse ~/mycontainer --tmp-path=/mnt/resource/blobfusetmp --config-file=/path/to/fuse_connection.cfg -o attr_timeout=240 -o entry_timeout=240 -o negative_timeout=120 -o allow_other --log-level=LOG_DEBUG

**Comment by vibhansa-msft** [comment]
with SCP I see an extra 'truncate' system call done for the file being copied, resulting into extra upload call. This will be fixed in the next release.

**Comment by takmnagayams** [comment]
@vibhansa-msft
>This will be fixed in the next release.
I really appreciate your help!

**Comment by vibhansa-msft** [comment]
This issue has been addressed in version 1.3.6. Kindly download the new package from below link or update the package using apt-get.
https://github.com/Azure/azure-storage-fuse/releases/tag/blobfuse-1.3.6
